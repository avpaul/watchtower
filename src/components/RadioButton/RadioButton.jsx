/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import PropsTypes from 'prop-types';
import './RadioButton.scss';

const CheckBox = props => {
  const { title, name, value, handleChange, options } = props;
  return (
    <div className="form-input">
      {title && (
        <label htmlFor={name} className="form-label">
          {title}
        </label>
      )}
      <div className="form-input-inline">
        <div className="radio-button">
          <input
            className="form-checkbox"
            id={name}
            name={name}
            type="checkbox"
            value={`${value}`}
            onClick={handleChange}
            checked={value}
            {...props}
          />
          {options}
        </div>
      </div>
    </div>
  );
};

CheckBox.propTypes = {
  name: PropsTypes.string.isRequired,
  title: PropsTypes.string,
  value: PropsTypes.bool.isRequired,
  handleChange: PropsTypes.func.isRequired,
  options: PropsTypes.string.isRequired
};

CheckBox.defaultProps = {
  title: ''
};

export default CheckBox;
