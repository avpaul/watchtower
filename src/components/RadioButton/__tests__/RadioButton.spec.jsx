import React from 'react';
import { shallow } from 'enzyme';
import RadioButton from '../RadioButton';

describe('Radio Button', () => {
  let wrapper;
  const props = {
    title: 'Form title',
    name: 'title',
    inputType: 'text',
    handleChange: jest.fn(),
    value: 'title',
    options: 'Certicicate is Exclusive',
    placeholder: 'title'
  };
  it('should render component correctly', () => {
    wrapper = shallow(<RadioButton {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
