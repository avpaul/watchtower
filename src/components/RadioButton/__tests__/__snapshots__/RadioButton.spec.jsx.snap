// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button should render component correctly 1`] = `
<div
  className="form-input"
>
  <label
    className="form-label"
    htmlFor="title"
  >
    Form title
  </label>
  <div
    className="form-input-inline"
  >
    <div>
      <input
        checked="title"
        className="form-radio"
        handleChange={[MockFunction]}
        id="title"
        inputType="text"
        name="title"
        onClick={[MockFunction]}
        options="Certicicate is Exclusive"
        placeholder="title"
        title="Form title"
        type="radio"
        value="title"
      />
      Certicicate is Exclusive
    </div>
  </div>
</div>
`;
